### ISSUE TITLE
Micronaut client does not generate proper XML for wrapped values

### ISSUE THREAD

**Issue by jmarshall5-chwy** [micronaut-projects/micronaut-core#5907]
### Expected Behavior

XML should be generated with a wrapped element, e.g.
```xml
<SimpleRequest>
    <id>1</id>
    <nestedItems>
        <nestedItem>
            <containerId>22</containerId>
            <someData>something</someData>
        </nestedItem>
        <nestedItem>
            <containerId>23</containerId>
            <someData>something else</someData>
        </nestedItem>
    </nestedItems>
</SimpleRequest>
```

### Actual Behaviour

Micronaut clients generate the following XML instead
```xml
<SimpleRequest>
    <id>1</id>
    <nestedItem>
        <containerId>22</containerId>
        <someData>something</someData>
    </nestedItem>
    <nestedItem>
        <containerId>23</containerId>
        <someData>something else</someData>
    </nestedItem>
</SimpleRequest>
```

### Steps To Reproduce

Check out the example program (here)[https://github.com/jmarshall5-chwy/micronaut_client_wrapped_xml_types_issue] and run

```shell
./gradlew test --tests "com.example.Xml_wrapperTest.doBasicXmlRequest" --info
```

It should run without error, but you will get a HttpClientResponseException as the server is not able to properly parse the XML.

See the README.MD in the project for more details, but the problem lies in the **BeanIntrospectionModule.java** Micronaut class. It attempts to make a reflection free encoder/decoder but it appears that Jackson explicitly depends on reflection to detect if the generator is doing XML or not.

This seems like it might be an issue with Jackson, in that perhaps the functions being used (_startWrappedValue_ & _finishWrappedValue in the ToXMLGenerator) should be part of the **GeneratorBase** class or **JSONGenertor** interface, but they don't.

I suspect this is a very low priority issue as this has probably been around since Micronaut 1.1 and it appears to only affect XML, but I at least wanted to document it so if others run into it they will not spend a day trying to figure it out.

### NOTE
This is only when encoding XML to be sent via a Micronaut client, it does not affect the Micronaut server when accepting request of properly formatted XML, e.g. you can use another client (like Spring) or just send a string version of the XML and it will be processed properly.

### Environment Information

Max OS X (Big Sur) - but shouldn't matter
Java 11 - but shouldn't matter

### Example Application

https://github.com/jmarshall5-chwy/micronaut_client_wrapped_xml_types_issue

### Version

2.5.11

**Comment by graemerocher** [comment]
I guess as a workaround you could replace `@Introspected` with `@ReflectiveAccess` on your data classes

**Comment by jmarshall5-chwy** [comment]
Thanks @graemerocher, I'm fairly new to Micronaut and hadn't seen `@ReflectiveAccess`, however; that annotation is for properties (not classes) and appears to require removing the `@Introspected` annotation (also implied by your comment). Unfortunately this ends up breaking validation checking. I've added a branch to my example app showing the changes I had to make to get it to work with that annotation.
Ideally we would want the server to work with `@Introspected` (and validation) as we would like the server to be a native app (via GrallVM). I suppose we could have different data structures for clients and server but that's adds a maintenance issue (code duplication).
Thanks for the info and possible work-around though.
